src/commands/utils/host.ts

```diff
@@ -3,20 +3,63 @@ import * as vscode from 'vscode';
 const meta = require('../../../package.json');
 
 export async function longRunning<T>(title: string, action: () => Promise<T>): Promise<T> {
-    const options = {
-        location: vscode.ProgressLocation.Notification,
-        title: title
-    };
-    return await vscode.window.withProgress(options, (_) => action());
+  const options = {
+    location: vscode.ProgressLocation.Notification,
+    title: title
+  };
+  return await vscode.window.withProgress(options, (_) => action());
 }
 
 export function getExtensionPath(): string | undefined {
-    const publisherName = `${meta.publisher}.${meta.name}`;
-    const vscodeExtensionPath = vscode.extensions.getExtension(publisherName)?.extensionPath;
+  const publisherName = `${meta.publisher}.${meta.name}`;
+  const vscodeExtensionPath = vscode.extensions.getExtension(publisherName)?.extensionPath;
 
-    if (!vscodeExtensionPath) {
-        vscode.window.showInformationMessage('No Extension path found.');
-        return;
-    }
-    return vscodeExtensionPath;
+  if (!vscodeExtensionPath) {
+    vscode.window.showInformationMessage('No Extension path found.');
+    return;
+  }
+  return vscodeExtensionPath;
 }
+
+export function getSASKey(
+  storageAccount: string,
+  storageKey: string,
+  generateSevenDaySas = false
```

Comment:
Consider making this something like `expiryDays` - then the caller is in control of what options to offer and this low-level function is not bound to the Periscope UI.